### PR TITLE
Feature/15.0/19853/_add_report_description

### PR DESCRIPTION
**Pull request by Jessica-BlueStingray**
### **[Add report description [19853]](https://erp.bluestingray.com/web#id=19853&view_type=form&model=project.task&action=378&menu_id=263)**

A description area was added to the Control Panel between the View name and the Search Area.
To test:
1. Create a fresh db
2. Install Sales (but can be tested in other modules too)
3. Turn on debug mode
4. Click on the debug button > Edit Action > Add a text to `view_description` > Save > Refresh page

https://user-images.githubusercontent.com/87671101/185498929-794a4734-1270-4d03-9149-ff7621804095.mp4

### If it doesn't show up at first, refresh the browser (Ctrl+R) one or more times, going to another view and coming back to the previous view also works..

### CODE REVIEW

File: addons/web/static/src/legacy/frontend/keyboard_navigation_mixin.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ odoo.define('web.KeyboardNavigationMixin', function (require) {
         },
 
         //--------------------------------------------------------------------------
+
```
Comment (dpsholtes) [inline addons/web/static/src/legacy/frontend/keyboard_navigation_mixin.js]:
```suggestion
```
Extra whitespace

File: addons/web/static/src/legacy/js/report/client_action.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ var ReportAction = AbstractAction.extend({
 
         this.action_manager = parent;
         this._title = options.display_name || options.name;
+//        this._viewDescription = options.view_description;
```
Comment (dpsholtes) [inline addons/web/static/src/legacy/js/report/client_action.js]:
```suggestion
```
Same as above

File: addons/web/static/src/legacy/js/views/abstract_controller.js [hunk 3]
```diff
@@ -423,6 +423,7 @@ var AbstractController = mvc.Controller.extend(ActionMixin, {
             actionMenus: this._getActionMenuItems(state),
             pager: this._getPagingInfo(state),
             title: this.getTitle(),
+//            viewDescription: this.getViewDescription(),
```
Comment (dpsholtes) [inline addons/web/static/src/legacy/js/views/abstract_controller.js]:
```suggestion
```
Same as above

File: addons/web/static/src/legacy/legacy_client_actions.js [hunk 4]
```diff
@@ -41,6 +41,7 @@ function registerClientAction(name, action) {
 
                 // always add user context to the action context
                 this.user = useService("user");
+//                this.viewDescription = useService("view_description");
```
Comment (dpsholtes) [inline addons/web/static/src/legacy/legacy_client_actions.js]:
```suggestion
```

File: addons/web/static/src/webclient/actions/action_dialog.js [hunk 5]
```diff
@@ -25,6 +25,7 @@ class ActionDialog extends Dialog {
         const action = actionProps && actionProps.action;
         this.actionType = action && action.type;
         this.title = "title" in this.props ? this.props.title : this.constructor.title;
+//        this.viewDescription = "view_description" in this.props ? this.props.view_description : this.constructor.view_description;
```
Comment (dpsholtes) [inline addons/web/static/src/webclient/actions/action_dialog.js]:
```suggestion
```

**Review by dpsholtes — CHANGES_REQUESTED** [review]
Untested on this pass. Just a couple questions and nitpicks

**Comment by traviswaelbro** [comment]
@Jessica-BlueStingray Let's keep this moving 🙏

File: addons/web/static/src/legacy/js/chrome/abstract_action.js [hunk 6]
```diff
@@ -145,7 +159,12 @@ const AbstractAction = Widget.extend(ActionMixin, {
             if ('title' in this.controlPanelProps) {
                 this._setTitle(this.controlPanelProps.title);
             }
+//            if ('view_description' in this.controlPanelProps) {
+//                this._setViewDescription(this.controlPanelProps
+//                .view_description);
+//            }
             this.controlPanelProps.title = this.getTitle();
+//            this.controlPanelProps.view_description = this.getViewDescription();
```
Comment (dpsholtes) [inline addons/web/static/src/legacy/js/chrome/abstract_action.js]:
```suggestion
            this.controlPanelProps.title = this.getTitle();
```
Sorry, I missed this one in my initial review

**Review by dpsholtes — CHANGES_REQUESTED** [review]
Functionally tested, looking good overall :+1:

Just one bit of commented out code to omit.

My only concern is this part of the requirements:
> Responsive design reclaims that whitespace for smaller screen widths, let’s hide the description as necessary.

I tested this out, and it does not appear to hide, and sometimes pushes the search bar off of the viewing area.

**Review by dpsholtes — APPROVED** [review]
Not retested. Looking good :+1:

**Review by toddythebody — APPROVED** [review]

**Review by traviswaelbro — APPROVED** [review]
Not retested but I did test previously ✅

**Comment by Jessica-BlueStingray** [comment]
@traviswaelbro @dpsholtes @toddythebody  This PR was rebased and is ready.

**Review by toddythebody — APPROVED** [review]

**Review by dpsholtes — APPROVED** [review]
Not retested

**Review by traviswaelbro — APPROVED** [review]

**Comment by traviswaelbro** [comment]
@Jessica-BlueStingray Leaving this for you to merge as you see fit

**Comment by Jessica-BlueStingray** [comment]
Waiting on Adam to see if the other PR with this code is going to be merged there or not. Still not really sure on what is the next move there.

**Comment by traviswaelbro** [comment]
Are you referring to MetricWise/odoo#20, where there was some discussion about `sed` commands?

**Pull request closed without merging**